packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2689,6 +2693,9 @@ export class SnapController extends BaseController<
 
       this.#updateApproval(pendingApproval.id, {
         loading: false,
+        ...(initialConnections !== undefined && initialConnections !== null
+          ? { connections: initialConnections }
+          : {}),
```

Comment:
```suggestion
        initialConnections: snap.initialConnections ?? {}
```

Wouldn't it be simpler to do something like this?

Reply:
Refactored and answered in: https://github.com/MetaMask/snaps/pull/2322#discussion_r1559458362